src/PrestaShopBundle/Bridge/AdminController/Listener/InitControllerListener.php

```diff
@@ -93,34 +82,21 @@ public function onKernelController(ControllerEvent $event): void
             return;
         }
 
+        /** @var BridgeControllerInterface $controller */
         $controller = $event->getController()[0];
-        $controllerNameLegacy = $event->getRequest()->attributes->get('_legacy_controller');
 
         if (!is_string(get_class($controller))) {
             return;
         }
 
-        $controller->php_self = get_class($controller);
         $this->context->smarty->assign('link', $this->context->link);
 
         $this->context->currentLocale = $this->localeRepository->getLocale(
             $this->context->language->getLocale()
         );
 
-        $controller->controllerConfiguration = $this->configurationFactory->create(
-            Tab::getIdFromClassName(
-                $controllerNameLegacy
-            ),
-            get_class($controller),
-            $controllerNameLegacy,
-            $controller->getTable()
-        );
-
-        $controller->multishop_context = $controller->controllerConfiguration->multishopContext;
-        $this->context->controller = $controller;
-
-        $this->setLegacyCurrentIndex($controller, $event->getRequest()->attributes->get('_legacy_controller'));
-        $this->initToken($controller, $event->getRequest());
+        $legacyControllerBridge = $controller->initLegacyControllerBridge($event->getRequest());
+        $this->context->controller = $legacyControllerBridge;
```

Comment:
I guess we'll have to update the PHPDoc of the Context class so that it can accept the `LegacyBridgeControllerInterface`